cognite/client/_api/extractionpipelines.py

```diff
@@ -0,0 +1,167 @@
+                >>> res = c.extraction_pipelines.retrieve_multiple(ids=[1, 2, 3])
+
+            Get assets by external id::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.extraction_pipelines.retrieve_multiple(external_ids=["abc", "def"], ignore_unknown_ids=True)
+        """
+        utils._auxiliary.assert_type(ids, "id", [List], allow_none=True)
+        utils._auxiliary.assert_type(external_ids, "external_id", [List], allow_none=True)
+        return self._retrieve_multiple(
+            ids=ids, external_ids=external_ids, ignore_unknown_ids=ignore_unknown_ids, wrap_ids=True
+        )
+
+    def list(self, limit: int = 25) -> ExtractionPipelineList:
+        """List extraction pipelines
+
+        Args:
+            limit (int, optional): Maximum number of ExtractionPipelines to return. Defaults to 25. Set to -1, float("inf") or None
+                to return all items.
+
+        Returns:
+            ExtractionPipelineList: List of requested ExtractionPipelines
+
+        Examples:
+
+            List ExtractionPipelines::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> ep_list = c.extraction_pipelines.list(limit=5)
+        """
+
+        return self._list(method="GET", limit=limit)
+
+    def create(
+        self, extractionPipeline: Union[ExtractionPipeline, List[ExtractionPipeline]]
+    ) -> Union[ExtractionPipeline, ExtractionPipelineList]:
+        """Create one or more extraction pipelines.
+
+        You can create an arbitrary number of extraction pipeline, and the SDK will split the request into multiple requests if necessary.
+
+        Args:
+            extractionPipeline (Union[ExtractionPipeline, List[ExtractionPipeline]]): Extraction pipeline or list of extraction pipelines to create.
+
+        Returns:
+            Union[ExtractionPipeline, ExtractionPipelineList]: Created extraction pipeline(s)
+
+        Examples:
+
+            Create new extraction pipeline::
+
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import ExtractionPipeline
+                >>> c = CogniteClient()
+                >>> extpipes = [ExtractionPipeline(name="extPipe1",...), ExtractionPipeline(name="extPipe2",...)]
+                >>> res = c.extraction_pipelines.create(extpipes)
+        """
+        utils._auxiliary.assert_type(extractionPipeline, "extraction_pipeline", [ExtractionPipeline, list])
+        return self._create_multiple(extractionPipeline)
+
+    def delete(self, id: Union[int, List[int]] = None, external_id: Union[str, List[str]] = None) -> None:
+        """Delete one or more extraction pipelines
+
+        Args:
+            id (Union[int, List[int]): Id or list of ids
+            external_id (Union[str, List[str]]): External ID or list of exgernal ids
+
+        Returns:
+            None
```

Comment:
if pipeline doesnt exist, does it ignore the return value?

Reply:
I'm not sure if I understood your question? It's the same behavior as for e.g. assets.delete.

Reply:
ah okay. though my question was: if the delete fails, should we not return an error or something. but if other services ignore that, would be fine to ignore here as well.

Reply:
It throws a `CogniteNotFoundError` in that case